src/ipc/ipc4/helper.c

```diff
@@ -392,10 +391,8 @@ int ipc_comp_disconnect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	if (!buffer)
 		return IPC4_INVALID_RESOURCE_ID;
 
-	irq_local_disable(flags);
-	list_item_del(buffer_comp_list(buffer, PPL_CONN_DIR_COMP_TO_BUFFER));
-	list_item_del(buffer_comp_list(buffer, PPL_CONN_DIR_BUFFER_TO_COMP));
-	irq_local_enable(flags);
+	pipeline_disconnect(src, buffer, PPL_CONN_DIR_COMP_TO_BUFFER);
+	pipeline_disconnect(sink, buffer, PPL_CONN_DIR_BUFFER_TO_COMP);
```

Comment:
The commit could use a few words why you do this? We get some additional debug and a bit dubious handling cache invalidation logic (in pipeline_disconnect), but what's the main motivation?